examples/src/bin/memlimit.rs

```diff
@@ -1,15 +1,14 @@
-#![no_std]
-#![no_main]
+#![cfg_attr(target_arch = "riscv32", no_std, no_main)]
 
 extern crate alloc;
 use alloc::vec;
 use core::hint::black_box;
 
-use nexus_rt::{println, Write};
+use nexus_rt::println;
 
 // use 16mb
 // alternatively, remove memlimit argument or otherwise set less than (slightly more than) 6mb and it will fail
-#[nexus_rt::main(memlimit(16))]
+#[nexus_rt::main(memlimit = 16)]
```

Comment:
This is a nice syntax.

Reply:
@sjudson I think while we're at it, two suggestions:
1. Rename `memlimit` to `stack_size`, as I assume this is what this argument configures. Memory size would usually mean heap size, so it creates confusion. E.g. jolt: https://github.com/a16z/jolt/blob/5d1b57cb1dac6b675e90e5a1db1271c44366a6b4/book/src/usage/troubleshooting.md?plain=1#L13
`memory_size` or `heap_size` will be available as a future reserved configuration.

2. Set it in bytes, not megabytes. Otherwise, we're dealing with conversions ourselves. I made it so it saturates, but it used to wrap/panic based on the build type.

Reply:
I decided not to apply it, will file a follow-up. I couldn't make the example work with both array or vec. Setting the limit to 16 **bytes** worked as well.